pkg/values/value.go

```diff
@@ -139,3 +152,38 @@ func FromDecimalValueProto(decStr string) (*Decimal, error) {
 	}
 	return NewDecimal(dec)
 }
+
+func createMapFromStruct(v any) (Value, error) {
```

Comment:
@patrick-dowell Looks good 👍 

At first glance I was going to propose using mapstructure for this, since that's what we use elsewhere but it seems mapstructure doesn't have an Encode method 🤷 

I do think we should handle float64 natively if there's a use case for it -- since you've had to workaround it, I'm assuming there is. Could you add a float62 type instead, rather than adding the convert method?

Reply:
This worked! Thanks!